mm2src/mm2_event_stream/src/streamer.rs

```diff
@@ -34,29 +35,31 @@ pub enum StreamerId {
     ForTesting(String),
 }
 
-impl fmt::Display for StreamerId {
+impl fmt::Display for StreamerIdInner {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
         match self {
-            StreamerId::Network => write!(f, "NETWORK"),
-            StreamerId::Heartbeat => write!(f, "HEARTBEAT"),
-            StreamerId::SwapStatus => write!(f, "SWAP_STATUS"),
-            StreamerId::OrderStatus => write!(f, "ORDER_STATUS"),
-            StreamerId::Task(task_id) => write!(f, "TASK:{task_id}"),
-            StreamerId::Balance(coin) => write!(f, "BALANCE:{coin}"),
-            StreamerId::TxHistory(coin) => write!(f, "TX_HISTORY:{coin}"),
-            StreamerId::FeeEstimation(coin) => write!(f, "FEE_ESTIMATION:{coin}"),
-            StreamerId::DataNeeded(data_type) => write!(f, "DATA_NEEDED:{data_type}"),
-            StreamerId::OrderbookUpdate(topic) => write!(f, "ORDERBOOK_UPDATE/{topic}"),
+            StreamerIdInner::Network => write!(f, "NETWORK"),
+            StreamerIdInner::Heartbeat => write!(f, "HEARTBEAT"),
+            StreamerIdInner::SwapStatus => write!(f, "SWAP_STATUS"),
+            StreamerIdInner::OrderStatus => write!(f, "ORDER_STATUS"),
+            StreamerIdInner::Task(task_id) => write!(f, "TASK:{task_id}"),
+            StreamerIdInner::Balance(coin) => write!(f, "BALANCE:{coin}"),
+            StreamerIdInner::TxHistory(coin) => write!(f, "TX_HISTORY:{coin}"),
+            StreamerIdInner::FeeEstimation(coin) => write!(f, "FEE_ESTIMATION:{coin}"),
+            StreamerIdInner::DataNeeded(data_type) => write!(f, "DATA_NEEDED:{data_type}"),
+            StreamerIdInner::OrderbookUpdate(topic) => write!(f, "ORDERBOOK_UPDATE/{topic}"),
             #[cfg(test)]
-            StreamerId::ForTesting(test_coin) => write!(f, "{test_coin}"),
+            StreamerIdInner::ForTesting(test_coin) => write!(f, "{test_coin}"),
         }
     }
 }
 
-impl fmt::Debug for StreamerId {
+impl fmt::Debug for StreamerIdInner {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result { write!(f, "{}", self) }
 }
 
+pub type StreamerId = Arc<StreamerIdInner>;
+
```

Comment:
this made `::new` calls so confusing. didn't know `StreamerId::new` actually invokes `Arc::new` at first glance.

I would prefer this to be a wrapping type: `struct StreamerId(Arc<StreamerInner>)` or `struct StreamerId { inner: Arc<StreamerInner> }`.